finetrainers/ltx_video/ltx_video_lora.py

```diff
@@ -174,8 +313,30 @@ def post_latent_preparation(
     width: int,
     patch_size: int = 1,
     patch_size_t: int = 1,
+    image_condition_dropout_p: float = 0.0,
+    image_condition_noise_scale: float = 0.0,
+    image_condition_noise_type: Optional[str] = "gaussian",
+    generator: Optional[torch.Generator] = None,
+    **kwargs,
 ) -> torch.Tensor:
+    assert image_condition_noise_type == "gaussian", "Only Gaussian noise is supported for now for LTX Video"
+
     latents = _normalize_latents(latents, latents_mean, latents_std)
+
+    # We know that the first frame is the image condition frame
+    if image_condition_dropout_p > 0.0:
+        # TODO(aryan): The zero-ing dropout is for testing based on ideas from CogVideoX. Since there is no source to
+        # refer to for LTX Video, this is experimental and should be unsupported if it does not work.
+        if random.random() < image_condition_dropout_p:
+            latents[:, :, 0].zero_()
+        else:
+            # Map from [0, 1] to [0, image_condition_noise_scale]
+            scale_factor = random.random() * image_condition_noise_scale
+            # :/ Because we don't have torch.randn_like
```

Comment:
Are we referring to [this](https://pytorch.org/docs/stable/generated/torch.randn_like.html)? Also, do we always have to add noise to the conditional latent? For Flux Control, we keep the conditional latent clean.

Reply:
Yes, torch.randn_like does not support using a `generator`, that's why I first create an empty tensor and then call `normal_` to create the gaussian noise.

Here, I tried making an educated guess that adding noise to the image would serve as a good regularizer because we don't have information on how it was trained. I haven't done enough experiments to see that it yields any benefits so I will be removing this

Reply:
For Flux as well, we _chose_ to not add any noise, but typically this should be experimented with so that the model can learn to pay better attention to control signals even if they are noisy. I have seen this being done atleast a few places now, so assumed it would make sense but unless I run a large 10k-50k step run, it would be hard to evaluate its effect

Reply:
Makes sense.

Reply:
So, IMO, it's okay to keep the argument for experimentation purposes.